### ISSUE TITLE
ctable.append does not accept a pandas.DataFrame

### ISSUE THREAD

**Issue by jseabold** [Blosc/bcolz#302]
Would be interested in making a pull request in this direction. Any thoughts? Is it silly just to convert a DataFrame to a ctable at the top of append?

**Comment by CarstVaartjes** [comment]
In our (still be open sourced) query engine, we do this:

    def df_to_bcolz(import_df, path_name, expected_len=None, debug=False):
        import_df = df_to_natural_name(import_df)
        if not os.path.exists(path_name):
            if not debug:
                print 'Creating ctable directory'
            if not expected_len:
                expected_len = len(import_df)
            ct = ctable.fromdataframe(import_df, expectedlen=expected_len, rootdir=path_name, mode='w')
        else:
            ct = ctable(rootdir=path_name, mode='a')
            ct_import = ctable.fromdataframe(import_df, expectedlen=len(import_df))
            ct.append(ct_import)
            del ct_import
        ct.flush()
        if debug:
            print 'Updated ctable: ', path_name

**Comment by FrancescAlted** [comment]
Yes, that would be a great addition.  And yes, using ctable.fromdataframe() would be an easy way to convert to a carray before appending.  Converting the dataframe to a numpy recarray would be an option too (looking at some benchmarks would be interesting before deciding).

**Comment by akloster** [comment]
I also naively tried to append a numpy array with the same row shape to a ctable, this fails with an IndexError. I haven't gotten that to work yet, except by iterating over the array and appending each row.

**Comment by CarstVaartjes** [comment]
my example code works for appending without an index error, but normally i do not use indexes in pandas except for monotonic selections. do you have a complex index?
